provision/rosa-cross-dc/cross-site-health-checks.sh

```diff
@@ -0,0 +1,126 @@
+    echo "  -n namespace: Kubernetes namespace"
+    exit 1
+}
+
+# Exit if no arguments are provided
+if [ $# -eq 0 ]; then
+    usage
+    exit 1
+fi
+
+# Parse input arguments
+while getopts ":d:u:p:s:n:c:h" opt; do
+  case ${opt} in
+    d ) domain=$OPTARG
+      ;;
+    u ) infinispan_user=$OPTARG
+      ;;
+    p ) infinispan_pwd=$OPTARG
+      ;;
+    s ) infinispan_url_suffix=$OPTARG
+      ;;
+    n ) namespace=$OPTARG
+      ;;
+    c ) expected_count=$OPTARG
+      ;;
+    h ) usage
+      ;;
+    \? ) echo "Invalid option: $OPTARG" 1>&2
+         usage
+         exit 1
+      ;;
+    : ) echo "Invalid option: $OPTARG requires an argument" 1>&2
+        usage
+        exit 1
+      ;;
+  esac
+done
+shift $((OPTIND -1))
+
+# Dependencies check
+for cmd in curl jq oc; do
+    if ! command -v $cmd &> /dev/null; then
+        echo "Error: Command $cmd is not available."
+        exit 1
+    fi
+done
+
+
+# Base URLs
+keycloak_lb_url="https://client.$domain"
+keycloak_site_a_url="https://primary.$domain"
+keycloak_site_b_url="https://backup.$domain"
+infinispan_rest_url="https://infinispan-external-runner-keycloak.apps.$infinispan_url_suffix"
+
+health_check() {
+    local url=$1
+    local parse_json=${2:-false}  # Optional second argument; default is 'false'
+
+    echo "Checking health for: $url"
+    if [ "$parse_json" = true ]; then
+        # If parsing JSON is requested, use jq to parse the response
+        response=$(curl -sk "$url")
+        echo "$response" | jq . 2>/dev/null
+        if [ ${PIPESTATUS[0]} -ne 0 ]; then
+            echo "Error parsing JSON from $url"
+            return 1
+        fi
+    else
+        # If parsing JSON is not requested, just display the raw response
+        curl -sk "$url" > /dev/null
+        if [ $? -eq 0 ]; then
+        	echo -e '\033[0;32m"HEALTHY"\033[0m'
+		else
+    		echo "Health check failed for: $url"
+		fi
+    fi
+    echo
+}
+
+echo "Verify the Keycloak App Health and Load Balancer health check"
+health_check $keycloak_lb_url/health
+health_check $keycloak_lb_url/lb-check
+
+echo "Verify the Load Balancer health check on Site A and Site B"
+health_check $keycloak_site_a_url/lb-check
+health_check $keycloak_site_b_url/lb-check
+
+echo "Verify the default cache manager health in external ISPN"
+health_check $infinispan_rest_url/rest/v2/cache-managers/default/health/status
+
+echo "Verify individual cache health"
+curl -u $infinispan_user:$infinispan_pwd -sk $infinispan_rest_url/rest/v2/cache-managers/default/health \
+ | jq 'if .cluster_health.health_status == "HEALTHY" and (all(.cache_health[].status; . == "HEALTHY")) then "HEALTHY" else "UNHEALTHY" end'
+echo
+
+echo "ISPN Cluster Distribution"
+curl -u $infinispan_user:$infinispan_pwd -sk $infinispan_rest_url/rest/v2/cluster\?action\=distribution \
+ | jq --argjson expectedCount $expected_count 'if map(select(.node_addresses | length > 0)) | length == $expectedCount then "HEALTHY" else "UNHEALTHY" end'
```

Comment:
I always get the following errors:

```bash
ISPN Cluster Distribution
jq: invalid JSON text passed to --argjson
Use jq --help for help with command-line options,
or see the jq manpage, or online docs  at https://stedolan.github.io/jq
```

Reply:
I still need to add docs on the usage of the script, but an example way of utilizing this would need to establish a session to the openshift cluster from your terminal session, alternatively we can think of using the oc login script internally.

```
./cross-site-health-checks.sh -d gh-keycloak-a-gh-keycloak-b-masked.keycloak-benchmark.com \
-u developer -p masked \
-s gh-keycloak-a.masked.openshiftapps.com \
-c 3 -n runner-keycloak
```

let me know if that works for you.

Reply:
I had missed the `-c 3` arg, adding this fixed the issue :slightly_smiling_face: . We need to add this flag to the "usage" string.

Reply:
thanks, thats a good catch Ryan. I have fixed it in my next commit.